consensus/src/chained_bft/chained_bft_smr.rs

```diff
@@ -281,21 +163,48 @@ impl<T: Payload> StateMachineReplication for ChainedBftSMR<T> {
             state_computer.sync_to_or_bail(initial_data.root_ledger_info());
         }
 
-        // Step 2 TODO: read from libradb instead of config
-        let epoch_mgr = Arc::new(EpochManager::new(
+        // Step 2 TODO: read validators from libradb instead of config
+        let executor = self
+            .runtime
+            .as_mut()
+            .expect("Consensus start: No valid runtime found!")
+            .executor();
+        let time_service = Arc::new(ClockTimeService::new(executor.clone()));
+
+        let (timeout_sender, timeout_receiver) =
+            channel::new(1_024, &counters::PENDING_PACEMAKER_TIMEOUTS);
+        let (self_sender, self_receiver) = channel::new(1_024, &counters::PENDING_SELF_MESSAGES);
+        let epoch_mgr = EpochManager::new(
             initial_setup.epoch,
             initial_setup.validator.clone(),
-        ));
-
-        // Step 3
-        self.start_epoch(
-            epoch_mgr,
+            self.config.take().expect("already started, config is None"),
+            time_service,
+            self_sender,
             initial_setup.network_sender,
-            initial_setup.network_events,
-            initial_setup.signer,
-            initial_data,
+            timeout_sender,
             txn_manager,
             state_computer,
+            self.storage.clone(),
+        );
+
+        // Step 3
+        let event_processor = epoch_mgr.start_epoch(initial_setup.signer, initial_data);
+
+        // TODO: this is test only, we should remove this
+        self.block_store = Some(event_processor.block_store());
```

Comment:
yeah, I agree, if a test needs this, it likely isn't testing at the right level... or we should expose some event callback